### PR TITLE
fix(nuxt): skip vue render when redirecting

### LINKED ISSUE #20696: Page setup and middleware running twice after SSR middleware redirect
### Environment

- Node Version:     v16.14.0
- Nuxt Version:     3.4.2
- Nitro Version:    2.3.3
- Package Manager:  yarn@1.22.10
- Builder:          vite
- User Config:      target, ssr, app, css, build, components, nitro, routeRules, vite, runtimeConfig, plugins, serverHandlers, modules
- Runtime Modules:  @pinia/nuxt@^0.4.0
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-3enzca-hrq7dz?file=pages/index.vue

Click in "Click here to hard reload A". Check logs to check when middleware and page setup run.

### Describe the bug

With SSR enabled, a request to path `/a` is redirected with a global middleware to path `/b`.

Expected behavior: global middleware runs and performs the redirect. It runs again and does not perform another redirect. Page B setup runs once.

Actual behavior: global middleware runs and performs the redirect. It runs again and does not perform another redirect. Page B setup runs. Global middleware runs again. Page B setup runs again.

### Logs

_No response_

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/20696

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

When we have an early response (for example, redirection), we can completely skip rendering the html for the page; we're not going to send it back. We can abort the rendering process with a targeted thrown/handled error.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/core/runtime/nitro/renderer.ts [hunk 1]
```diff
@@ -247,6 +247,9 @@ export default defineRenderHandler(async (event): Promise<Partial<RenderResponse
   }
 
   const _rendered = await renderer.renderToString(ssrContext).catch(async (error) => {
+    // We use error to bypass full render if we have an early response we can make
+    if (ssrContext._renderResponse && error.message === 'skipping render') { return {} as ReturnType<typeof renderer['renderToString']> }
```
Comment (pi0) [inline packages/nuxt/src/core/runtime/nitro/renderer.ts:251]: (please hold on to this. returning empty is probably unsafe while checking https://github.com/nuxt/nuxt/issues/21414)

Comment (danielroe) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:251]: We do not return an empty response - we return the value of `ssrContext._renderResponse` below on L260.

Comment (danielroe) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:251]: do you still want to defer this PR?

Comment (pi0) [reply packages/nuxt/src/core/runtime/nitro/renderer.ts:251]: Seems good if we are not returning an empty object to the nitro renderer :)

**Pull request merged**